src/cms.c

```diff
@@ -31,66 +43,84 @@ void CMS_Destroy(CMSketch *cms) {
     CMS_FREE(cms);
 }
 
-void CMS_IncrBy(CMSketch *cms, const char *item, size_t value) {
+void CMS_IncrBy(CMSketch *cms, const char *item, size_t strlen, size_t value) {
     assert(cms);
     assert(item);
 
     for(size_t i = 0; i < cms->depth; ++i) {
-        size_t hash = XXH64(item, strlen(item), i);
+        uint32_t hash = MurmurHash2(item, strlen, i);
         cms->array[(hash % cms->width) + (i * cms->width)] += value;
     }
     cms->counter += value;
 }
 
-size_t CMS_Query(CMSketch *cms, const char *item) {
+size_t CMS_Query(CMSketch *cms, const char *item, size_t strlen) {
     assert(cms);
     assert(item >= 0);
 
-    size_t temp = 0, res = ~0;
+    size_t temp = 0, res = -1;
 
     for(size_t i = 0; i < cms->depth; ++i) {
-        size_t hash = XXH64(item, strlen(item), i);
+        uint hash = MurmurHash2(item, strlen, i);
         temp = cms->array[(hash % cms->width) + (i * cms->width)];
-        if(temp < res) res = temp;
+        if(temp < res) {
```

Comment:
Formatting. I suggest you use `clang-format` (already an option in vscode)

Reply:
Nice one.